momentGW/util.py

```diff
@@ -473,7 +473,12 @@ def _fallback():
 
     # Reshape and transpose the output
     ct = ct.reshape(shape_ct, order="A")
-    c = ct.transpose(order_ct)
+    if ct.flags.f_contiguous:
+        c = np.asfortranarray(ct.transpose(order_ct))
+    elif ct.flags.c_contiguous:
+        c = np.ascontiguousarray(ct.transpose(order_ct))
+    else:
+        c = ct.transpose(order_ct)
```

Comment:
can you explain these changes? does it just intend the output to be contiguous?

my original code allows `ct` to be in any contiguous order, and then doesn't therein constrain the contiguity of `c`. Sometimes a transpose can just be equivalent to reading the array with the opposite contiguity, if the transpose indices are $(N-1, N-2, ..., 0)$. This means that in the worst case the new code will require an entire extra copy of `c`.

if we want to constrain the contiguity of the output, I think it would be better to just do
```
ct = ct.reshape(shape_ct, order="A")
c = ct.transpose(order_ct)
c = np.asarray(c, order="A")
```

Reply:
This doesn't quite work as the contiguity has already been changed with the transpose, so `order="A"` just keeps it as F contiguous

Reply:
I can't seem to see it in another PR, but just want to check this isn't change in another PR.

Reply:
i've not changed it, just revert this on this PR